lib/TickLib.sol

```diff
@@ -174,33 +174,49 @@ library LeafLib {
 library TickLib {
 
   function eq(Tick tick1, Tick tick2) internal pure returns (bool) {
-    return Tick.unwrap(tick1) == Tick.unwrap(tick2);
+    unchecked {
+      return Tick.unwrap(tick1) == Tick.unwrap(tick2);
+    }
   }
 
   function inRange(Tick tick) internal pure returns (bool) {
-    return Tick.unwrap(tick) >= MIN_TICK && Tick.unwrap(tick) <= MAX_TICK;
+    unchecked {
+      return Tick.unwrap(tick) >= MIN_TICK && Tick.unwrap(tick) <= MAX_TICK;
+    }
   }
 
   function fromLogPrice(int logPrice, uint tickScale) internal pure returns (Tick) {
-    // Do not force logPrices to fit the tickScale (aka logPrice%tickScale==0)
-    // Round all prices down (aka cheaper for taker)
-    int tick = logPrice / int(tickScale);
-    if (logPrice < 0 && logPrice % int(tickScale) != 0) {
-      tick = tick - 1;
+    unchecked {
+      // Do not force logPrices to fit the tickScale (aka logPrice%tickScale==0)
+      // Round all prices down (aka cheaper for taker)
+      int tick = logPrice / int(tickScale);
+      if (logPrice < 0 && logPrice % int(tickScale) != 0) {
+        tick = tick - 1;
+      }
+      return Tick.wrap(tick);
     }
-    return Tick.wrap(tick);
   }
 
+  // Utility for tests&unpacked structs, less gas-optimal
   function tickFromBranch(uint tickPosInLeaf,Field level0, Field level1, Field level2, Field level3) internal pure returns (Tick) {
     unchecked {
-      uint utick = tickPosInLeaf |
-        ((BitLib.ctz(Field.unwrap(level0)) |
-          (BitLib.ctz(Field.unwrap(level1)) |
-            (BitLib.ctz(Field.unwrap(level2)) |
-              uint((int(BitLib.ctz(Field.unwrap(level3)))-LEVEL3_SIZE/2) << LEVEL2_SIZE_BITS)) 
+      LocalPacked local;
+      local = local.tickPosInLeaf(tickPosInLeaf).level0(level0).level1(level1).level2(level2).level3(level3);
+      return tickFromLocal(local);
+    }
+  }
+
+  function tickFromLocal(LocalPacked local) internal pure returns (Tick) {
```

Comment:
These return the *best* tick on that branch, right? If so, would be good if the name reflected this, eg, `bestTickFromBranch`.

Reply:
see #544